### PR TITLE
VAOS service error middleware

### PR DESCRIPTION
**Pull request by kreek**
## Description of change
Moves VAOS service error handling to middleware.

## Testing done
unit tests

## Testing planned
staging integrations tests

#### Unique to this PR

- [ ] _Replace this line with individual tasks unique to your PR_

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/vaos/app/services/vaos/middleware/response/errors.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module VAOS
+  module Middleware
+    module Response
+      class Errors < Faraday::Response::Middleware
+        def on_complete(env)
+          return if env.success?
+
+          caller = "VAOS.#{caller_locations(14, 1)[0].label}"
```
Comment (kreek) [inline modules/vaos/app/services/vaos/middleware/response/errors.rb]: this gets the calling method e.g. VAOS.get_systems don't know if there's a better way to get the class and method that called it?

Comment (saneshark) [reply modules/vaos/app/services/vaos/middleware/response/errors.rb]: This is kind of nice, but is it really necessary? I would think the backtrace should tell us what we need.

File: modules/vaos/app/services/vaos/middleware/response/errors.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          case env.status
+          when 403
+            raise Common::Exceptions::BackendServiceException.new(
+              'VAOS_403',
+              source: caller
+            )
+          when 500
+            raise Common::Exceptions::BackendServiceException.new(
+              'VAOS_502',
+              source: caller
```
Comment (saneshark) [inline modules/vaos/app/services/vaos/middleware/response/errors.rb]: these are going to get pretty unruly fast... do we want a range of 500s? Also does breakers still have an opportunity to do it's thing or does this middleware circumvent that?

Comment (kreek) [reply modules/vaos/app/services/vaos/middleware/response/errors.rb]: it's after breakers in the chain

**Comment by saneshark** [comment]
This should probably be merged for production testing to begin. Gives much more robust error handling and makes Sentry debugging easier.

**Pull request merged**